src/driver-hidpp20.c

```diff
@@ -344,6 +344,14 @@ hidpp20drv_read_led(struct ratbag_led *led)
 		hidpp20drv_read_led_8070(led, drv_data);
 	else if (drv_data->capabilities & HIDPP_CAP_LED_SW_CONTROL_1300)
 		hidpp20drv_read_led_1300(led, drv_data);
+	else
+		return;
+
+	ratbag_led_set_mode_capability(led, RATBAG_LED_ON);
+	ratbag_led_set_mode_capability(led, RATBAG_LED_CYCLE);
```

Comment:
Monochrome LEDs don't support the cycle mode since there's no color to cycle between. Moreover not all LEDs even support the basic on mode (some can only be put in breathing mode). For the monochrome I would rather fetch it from the mouse in hidpp20drv_read_led_1300.

Reply:
I moved this to the respected read_lead_XXXX and dropped the cycle bit. For anything more specific it's prob. better to do a follow-up patch because it seems we need some .device file entries for this, don't we?

Reply:
Actually no, at least not for the monochrome LEDs (and as far as I know all RGB LEDs support the 4 effects), since they advertise their supported effects through the 0x1300 feature. The caps attribute of struct hidpp20_led_sw_ctrl_led_info is a bitfield indicating which effects are supported.